keylime/tpm/tpm_main.py

```diff
@@ -1331,12 +1387,14 @@ def sim_extend(self, hashval_1, hashval_0=None, hash_alg=None):
             hashval_0 = self.START_HASH(hash_alg)
 
         # compute expected value  H(0|H(data))
-        extendedval = self.hashdigest(
-            codecs.decode(hashval_0, "hex_codec")
-            + codecs.decode(self.hashdigest(hashval_1.encode("utf-8"), hash_alg), "hex_codec"),
+        hdata = self.hashdigest(hashval_1.encode("utf-8"), hash_alg)
+        assert hdata is not None
```

Comment:
What can go wrong here that the value might be None?

Reply:
https://github.com/mdrocco/keylime/blob/71b33a96d084321be7dcb0ef78ba4c3a1edf8a66/keylime/tpm/tpm_main.py#L1395

`codecs.decode` does not accept `None` for the `obj` parameter.

Reply:
Yeah, but can `self.hashdigest(...)` return None?

Reply:
I don't understand your question, the type checker is making the point about `codecs.decode`.

Reply:
Ah ok, then keep the asserts. On another note I think it is fine to just assign it to same variable instead of using a new one or does the type checker than complain?

Reply:
I am not sure what you mean by "assign it to the same variable", but as long as we are not re-typing the variable (which I second the checker if it complains about, as it makes the code unnecessarily obscure) I am happy to hear any simplification you have to propose.